src/lib/Libutil/misc_utils.c

```diff
@@ -1835,9 +1837,21 @@ create_query_file(void)
 {
 	FILE *f;
 	char filename[MAXPATHLEN + 1];
-	snprintf(filename, sizeof(filename), "%s/.pbs_last_query_%d", TMP_DIR, getuid());
+	uid_t usid = getuid();
+#ifdef WIN32
+	LPSTR win_sid = NULL;
+	if (!ConvertSidToStringSid(usid, &win_sid)) {
+		fprintf(stderr, "qstat: failed to convert SID to string with error=%d\n", GetLastError());
+		return;
+	}
+	snprintf(filename, sizeof(filename), "%s\\.pbs_last_query_%s", TMP_DIR, win_sid);
+	LocalFree(win_sid);
+#else
+	snprintf(filename, sizeof(filename), "%s/.pbs_last_query_%d", TMP_DIR, usid);
+#endif // WIN32
 	f = fopen(filename, "w");
-	fclose(f);
+	if (f != NULL)
+		fclose(f);
```

Comment:
I'm not sure I understand what is being accomplished by opening and closing the file.  If it is there, you'll open it and then immediately close it without doing anything.  If it is not there, you'll fail to open it and move on.  Either way, nothing has been done.

Reply:
We are creating this temporary file in order to get the timestamp of the last qstat command that was executed by a user.
In delay_query() function, we are using stat() function to check the last modified time of that particular file to see difference of time between two consecutive qstats.